CMakeLists.txt

```
@@ -1,88 +1,58 @@
+# using:
+#
+# `-DJULIA_BINDIR=/path/to/.../julia/bin`
+#
+# If you are unsure of the location of this folder, you can access
+# it from within the REPL using:
+#
+# `println(Sys.BINDIR)`
+#
+# ### Options ##
+#
+# jluna_DEVELOPER_MODE      : enable building test and benchmark executables. Off by default
+# BUILD_TESTING             : build jluna_test, as CTest. On by default
+# BUILD_BENCHMARK           : build jluna_benchmark. On by default
+#
+
+project(jluna VERSION 0.8.4 LANGUAGES CXX)
+
+include(cmake/project-is-top-level.cmake)
+if(PROJECT_IS_TOP_LEVEL)
+    option(jluna_DEVELOPER_MODE "Enable developer mode" OFF)
 endif()
 
-### COMPILER SUPPORT ###
+### Find Julia ###
 
-if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION STRGREATER_EQUAL "12.0.0")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-volatile")
-elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION STRGREATER_EQUAL "10.0.0")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fconcepts -Wno-volatile")
-elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CMAKE_CXX_COMPILER_VERSION STRGREATER_EQUAL "19.20")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-else()
-    message(FATAL_ERROR "Currently, the only supported compilers are G++10 (or higher), Clang-12 (or higher) and MSVC 19.30 (experimental)")
-endif()
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/find")
+find_package(Julia 1.7.0 REQUIRED)
 
-### JULIA ###
+### Configure Files ###
 
-if (WIN32)
-    if (NOT DEFINED ENV{JULIA_PATH})
-        message(FATAL_ERROR "Cannot determine location of julia image. Before running cmake, please manually set the environment variable JULIA_PATH using:\n\tset JULIA_PATH=\"C:/path/to/your/.../julia-1.7.2\"\nIf you are unsure of the location of your julia image, you can access the path from within the julia REPL using\n    println(joinpath(Sys.BINDIR, \"..\"))")
-    endif()
-else()
-    # access julia through julia environment variable
-    if (NOT DEFINED ENV{JULIA_PATH})
-        execute_process(
-                COMMAND julia -e "println(joinpath(Sys.BINDIR, \"..\"))"
-                OUTPUT_VARIABLE JULIA_PATH_LOCAL)
-
-        if ("${JULIA_PATH}" STREQUAL "")
-            message(FATAL_ERROR "Cannot determine location of julia image. Before running cmake, please manually set the environment variable JULIA_PATH using\n    export JULIA_PATH=/path/to/your/.../julia\nIf you are unsure of the location of your julia image, you can access the path from within the julia REPL using\n    println(joinpath(Sys.BINDIR, \"..\"))
-    For more information, visit https://github.com/Clemapfel/jluna/blob/master/README.md#troubleshooting")
-        endif()
-
-        set(ENV{JULIA_PATH} ${JULIA_PATH_LOCAL})
-    endif()
-endif()
+file(READ include/jluna.jl JULIA_SOURCE)
+set(C_ADAPTER_PATH ${CMAKE_INSTALL_PREFIX})
+configure_file("${CMAKE_SOURCE_DIR}/.src/include_julia.inl.in" "${CMAKE_SOURCE_DIR}/.src/include_julia.inl" @ONLY)
```

Don't output to the source directory. This will dirty the project.

```suggestion
configure_file(.src/include_julia.inl.in .src/include_julia.inl @ONLY)
```